mgmt/rest/task.go

```diff
@@ -66,6 +66,7 @@ type task struct {
 	FailedCount        uint                    `json:"failed_count,omitempty"`
 	LastFailureMessage string                  `json:"last_failure_message,omitempty"`
 	State              string                  `json:"task_state"`
+	MaxFailure         uint                    `json:"max_failure"`
```

Comment:
Do we want this to be uint? I know as part of #967 we were thinking of using -1 (or I guess anything <0) to represent infinite retries. If that is the case we might just want this to be int?